### PR TITLE
MM-28338 Fix Combined system messages 

### PR DESCRIPTION
**Pull request by Safouene1**
#### Summary

[OLD PR (before monorepo) LINK ](https://github.com/mattermost/mattermost-webapp/pull/12088)
Fix the chronological order of combined system messages and remove the priority based sorting as discussed in [here](https://github.com/mattermost/mattermost-server/issues/21926)

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/21926

JIRA: https://mattermost.atlassian.net/browse/MM-28338

#### Related Pull Requests
none

#### Screenshots
Steps to reproduce:
*Create a new public channel.
*Add a user to the channel.
*Post a random message to the channel
*Use the "Manage Member" menu to remove the user
*Use the "Manage Member" menu to add the user back into the channel

|  before  |  after  |
|----|----|
|image|image_2023-01-28_003512141|

#### Release Note
```
Fixed a bug where combined system messages are not in chronological order
```
```release-note
Fixed a bug where combined system messages are not in chronological order
```

### CODE REVIEW

**Comment by Safouene1** [comment]
Hey @hmhealey , Sorry for pinging you but you're the only one informed about everything about this PR.

While working on tests i discovered the following : 
Currently, in the master branch, getting usernames in the combineActivity feature is broken and does not work. However, in this PR, it now works as intended and the issue has been fixed 
on line 333 in post lists

`    messageData.push({postType, userIds: [...usernames, ...userIds], actorId: actorId[0]});
`

However, I noticed that in messageData, we are sending both usernames and userIds to the userIds array. I believe this is unnecessary as the old implementation had broken usernames (empty array), so only IDs were being sent. Additionally, I looked into the combined_system_message and found that it can work without usernames just fine.

I suggest removing usernames from the userIds array (userIds: [...userIds]) to simplify the code and improve its efficiency. Please let me know your thoughts on this.

**Comment by hmhealey** [comment]
> I suggest removing usernames from the userIds array (userIds: [...userIds]) to simplify the code and improve its efficiency. Please let me know your thoughts on this.

Yeah, that sounds good to me. I think we used to pass both usernames and user IDs since older system messages were inconsistent of which ones were included in the post props, but since you have to go really far back to see those posts now, I think it's fine to not pass both of them. I definitely appreciate all the work you're doing to help clean this code up 🙂

**Comment by Safouene1** [comment]
I've updated tests and added a filter to null values in AllUserIDS and AllUsernames.

**Comment by hmhealey** [comment]
Cool, let me know when this is ready for review again then

**Comment by Safouene1** [comment]
Hey ! I deleted the 2 tests because the combine system messages do not handle those operations anymore. 
Should be good now.

**Comment by hmhealey** [comment]
Cool, I'll review that shortly. I had to merge master again to get the CI tests unblocked, so hopefully that's okay with you. One day, we'll stop changing those constantly

File: webapp/channels/src/packages/mattermost-redux/src/utils/post_list.ts [hunk 1]
```diff
@@ -396,69 +359,45 @@ function extractUserActivityData(userActivities: any) {
...
+        // When combining removed posts, the actorId does not need to be the same for each post.
+        // All removed posts will be combined regardless of their respective actorIds.
+        const isRemovedPost = post.type === Posts.POST_TYPES.REMOVE_FROM_CHANNEL;
+        const userId = isUsersRelatedPost(postType) ? post.props.addedUserId || post.props.removedUserId : '';
+        const username = isUsersRelatedPost(postType) ? post.props.addedUsername || post.props.removedUsername : '';
+        const prevPost = userActivities[userActivities.length - 1];
+        const isSamePostType = prevPost && prevPost.postType === post.type;
+        const isSameActor = prevPost && prevPost.actorId[0] === post.user_id;
+
+        if ((isSamePostType && isSameActor && prevPost) || (isSamePostType && prevPost && isRemovedPost)) {
```
Comment (hmhealey) [inline webapp/channels/src/packages/mattermost-redux/src/utils/post_list.ts:387]:
```suggestion
        if (prevPost && isSamePostType && (isSameActor || isRemovedPost)) {
```

**Review by hmhealey — CHANGES_REQUESTED** [review]
Thanks for updating all those tests. I have one small suggestion to make an if statement a bit easier to read, but otherwise, this looks good to me

**Comment by hmhealey** [comment]
Looks like the tests are also failing because an old snapshot needs to be removed

**Comment by Safouene1** [comment]
> Looks like the tests are also failing because an old snapshot needs to be removed

Sorry about those , I forgot about deleting them when I deleted the 2 tests. Will update asap

**Review by hmhealey — APPROVED** [review]

**Review by koox00 — APPROVED** [review]
Thanks @Safouene1 💯

**Comment by hmhealey** [comment]
@Safouene1 We're also going to want to make these changes on our mobile app as well. If we open that as another help wanted ticket, would you be interested in working on that as well? If not, that's totally fine too

**Comment by Safouene1** [comment]
> @Safouene1 We're also going to want to make these changes on our mobile app as well. If we open that as another help wanted ticket, would you be interested in working on that as well? If not, that's totally fine too

I would love to give it a try

**Comment by Safouene1** [comment]
Great ! Please assign me or mention me in the new ticket.
I will setup the environment and development in the weekend

**Review by furqanmlk — APPROVED** [review]
Working as expected

**Before**

image

**After**

image

**Pull request merged**